buildSrc/src/main/java/com/gradle/publish/DownloadMavenArtifactsAndPublishToGradlePluginPortal.java

```diff
@@ -0,0 +1,311 @@
+    downloadArtifact(client, artifactUrlPrefix + "-javadoc.jar", libsDir.resolve(artifactId + "-" + version + "-javadoc.jar"));
+    downloadArtifact(client, artifactUrlPrefix + "-javadoc.jar.asc", libsDir.resolve(artifactId + "-" + version + "-javadoc.jar.asc"));
+  }
+
+  /**
+   * Validate that minimum publish content contains: pom, jar, sources, javadoc, module and their signatures.
+   */
+  private void validatePublishContent() {
+    LOGGER.info("ValidatePublishContent artifacts");
+    Map<PublishArtifact, File> artifacts = collectArtifacts();
+    artifacts.forEach((artifact, file) -> LOGGER.info("Publish artifact {} from file {}", artifact.encode(), file));
+
+    Set<String> actualArtifactTypes = artifacts.keySet().stream().map(PublishArtifact::getType).collect(Collectors.toSet());
+
+    Set<String> missingArtifactTypes = REQUIRED_ARTIFACT_TYPES.stream()
+      .filter(type -> !actualArtifactTypes.contains(type))
+      .collect(Collectors.toSet());
+
+    if (!missingArtifactTypes.isEmpty()) {
+      String missingArtifactTypesString = String.join(", ", missingArtifactTypes);
+      throw new GradleException("Missing required artifact types: " + missingArtifactTypesString);
+    }
+  }
+
+  public void downloadArtifact(HttpClient client, String artifactUrl, Path... destinationPaths) throws InterruptedException {
+    LOGGER.info("Downloading {}", artifactUrl);
+    HttpRequest request = HttpRequest.newBuilder()
+      .uri(URI.create(artifactUrl))
+      .header("Authorization", "Bearer " + mavenAuthorizationToken)
+      .GET()
+      .build();
+    HttpResponse<byte[]> response;
+    try {
+      response = client.send(request, HttpResponse.BodyHandlers.ofByteArray());
+    } catch (IOException e) {
+      throw new GradleException("Failed to download content from '" + artifactUrl + "'. Error: " + e.getMessage(), e);
+    }
+    if (response.statusCode() != 200) {
+      throw new GradleException("Failed to download content from '" + artifactUrl + "'. HTTP status: " + response.statusCode());
+    }
+    byte[] content = response.body();
+    for (Path destinationPath : destinationPaths) {
+      Path rootDir = getProject().getRootDir().toPath();
+      Path destination = rootDir.resolve(destinationPath);
+      LOGGER.info("Saving to {}", destination);
+      try {
+        Files.createDirectories(destination.getParent());
+        Files.write(destination, content);
+      } catch (Exception e) {
+        throw new GradleException("Failed to save content to '" + destination + "'. Error: " + e.getMessage(), e);
+      }
+    }
+  }
+
+  private void executePublishTask() {
+    try {
+      LOGGER.info("Invoke 'publishPlugins' task (validationOnly: {})", validationOnly);
+      PublishTask publishTask = (PublishTask) getProject().getTasks().getByName("publishPlugins");
+      Method executeTask = PublishTask.class.getDeclaredMethod("executeTask");
+      executeTask.setAccessible(true);
```

Comment:
Just to clarify: now that we have an explicit dependency on the publish plugin, do we still need access the method through reflection?

Reply:
If it is possible let's create a ticket to tackle that later